tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -319,6 +319,15 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( @"[Blocking] Foo Bar() { var ct = new CancellationToken(); return Foo( 1,""test"" ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void ParenthesizedAnonymousCreationLambda() {

```

Comment:
Are parenthesized lambdas and anonymous objects creations actually directly related that we're testing them together like this?

Reply:
Pretty much the only ~500 uses of anonymous object creations in the LMS are in parenthesized lambda expressions to do `m_logger.Debug()` or similar operations

Reply:
That's true, and we do need to test combinations of things, but this is the only test for parenthesized lambdas, and the only test for anonymous object creations. I think we could use more tests for both so we're describing our target behaviour, and to show that they aren't actually tied together (while anonymous object creations might mostly be used in logging, we use parenthesized lambdas in many places).

Not going into it in this comment, but maybe lets brainstorm a number of examples of parenthesized lambda usages and anonymous object creations, and go from there?

(Also, I see something more like 2000+ hits. Did you mean something else by ~500: https://search.d2l.dev/search?project=lms&full=%22new+%7B%22&defs=&refs=&path=-*Test*&hist=&type=csharp&xrd=&nn=1&n=25&start=2325).

Reply:
Yeah the 500 was a different search I made